aws_lambda_powertools/utilities/data_classes/common.py

```diff
@@ -28,6 +28,49 @@ def __iter__(self) -> Iterator:
     def __len__(self) -> int:
         return len(self._data)
 
+    def __str__(self) -> str:
+        return str(self._str_helper())
+
+    def _str_helper(self) -> Dict[str, Any]:
+        """
+        Recursively get a Dictionary of DictWrapper properties primarily
+        for use by __str__ for debugging purposes.
+
+        Will remove "raw_event" properties, and any defined by the Data Class
+        `_str_sensitive_data` list field.
+        This should be used in case where secrets, such as access keys, are
+        stored in the Data Class but should not be logged out.
+        """
+        properties = self._properties()
+        sensitive_properties = ["raw_event"]
+        try:
+            sensitive_properties.extend(self.__getattribute__("_str_sensitive_data"))
+        except AttributeError:
+            pass
```

Comment:
```suggestion
        properties = self._properties()
        sensitive_keys = ["raw_event"]
        if hasattr(self, "_str_sensitive_data"):
            sensitive_keys.extend(self._str_sensitive_data)
```

I think `sensitive_keys` is more intuitive and using `if` instead of a `try/catch` block makes the code more readable for others.

Reply:
> using if instead of a try/catch block makes the code more readable for others

I agree - I'll rebase this in.

> I think sensitive_keys is more intuitive

I used properties as we use the `@properties` decoration, and wanted to be clear it is printing properties instead of keys from the `raw_event` dict.

Reply:
>using if instead of a try/catch block makes the code more readable for others
> I agree - I'll rebase this in.
Done.

Reply:
I've also made sure all places use `_str_sensitive_properties` (instead of `_str_sensitive_data`) for consistency.

Reply:
Updated again to `_sensitive_properties` based off feedback.